src/NUnitEngine/nunit.engine.api/ITestAgentInfo.cs

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Charlie Poole, Rob Prouse and Contributors. MIT License - see LICENSE.txt
+
+using System.Collections.Generic;
+
+namespace NUnit.Engine.Services
+{
+    /// <summary>
+    /// Objects implementing ITestAgentInfo by classes that can return information about
+    /// available agents. The interface is used by runners in order to provide information
+    /// to the user or to allow selecting agents.
+    /// </summary>
+    public interface ITestAgentInfo
+    {
+        /// <summary>
+        /// Gets a list containing <see cref="TestAgentInfo"/> for all available agents.
+        /// </summary>
+        IList<TestAgentInfo> GetAvailableAgents();
+
+        /// <summary>
+        /// Gets a list containing <see cref="TestAgentInfo"/> for any available agents,
+        /// which are able to handle the specified package.
+        /// </summary>
+        /// <param name="package">A TestPackage</param>
+        /// <returns>
+        /// A list of suitable agents for running the package or an empty
+        /// list if no agent is available for the package.
+        /// </returns>
+        IList<TestAgentInfo> GetAgentsForPackage(TestPackage package);
```

Comment:
Is there value in having two separate methods for this interface instead of just one with an optional TestPackage parameter.  In practice GetAgentsForPackage is always calling GetAvailableAgents and then filtering the result, so it could be easily changed to just not filter unless a package is provided.

Reply:
I wanted to emphasize the key difference, i.e. that GetAvailableAgents is stable for the test run. Probably should have made it a lazy property in order to signal that. Does that sound better?

Reply:
Made that change, initializing the property in the `StartService` method. Looks better to me.